04_testing/outlier_checks.R

```diff
@@ -897,3 +898,365 @@ write_csv(
     "2022_distribution_min_2nd_max_pins.csv"
   )
 )
+
+###############################################
+#### RELATIONSHIP WITH DISAGG AND OUTLIERS ####
+###############################################
+
+# get labels for cutting bins
+log_labs <- sapply(
+  0:15,
+  function(x) {
+    paste(
+      scales::comma(exp(x)),
+      scales::comma(exp(x + 1)),
+      sep = " - "
+    )
+  }
+)
+
+df_outlier_pop <- df_outliers %>%
+  mutate(
+    log_pop = log(affected_population),
+    log_pop_bins = cut(
+      log_pop,
+      breaks = 0:16,
+      labels = log_labs,
+      include.lowest = TRUE
+    )
+  ) %>%
+  group_by(
+    log_pop_bins
+  ) %>%
+  summarize(
+    is_any_outlier = is_any_outlier / n(),
```

Comment:
in my computer the dataset stays the same size with the same 16 percentages being repeated, added sum to is_any_outlier then it reduced to the 16 bins, it's probably more efficient if there are no duplication

Reply:
Good catch!